lib/open_food_network/enterprise_injection_data.rb

```diff
@@ -22,16 +22,21 @@ def earliest_closing_times
     end
 
     def shipping_method_services
-      @shipping_method_services ||= CacheService.cached_data_by_class("shipping_method_services",
-                                                                      Spree::ShippingMethod) do
+      @shipping_method_services ||= CacheService.cached_data_by_class(
+        "shipping_method_services_#{@enterprise_ids.hash}",
```

Comment:
Ah, that's a good point. My _hope_ here would be that caching wouldn't even be needed if fetching and processing the data is fast enough - currently, `CacheService.cached_data_by_class` performs an extra DB query just to generate the cache key `cached_class.maximum(:updated_at).to_f`; on my machine it normally takes a few ms but has taken up to 45ms (though maybe it would be more consistent with an index on updated_at), but it's a similar time to actually pulling all the rows anyway:

<img width="898" alt="Screenshot 2024-08-13 at 14 53 19" src="https://github.com/user-attachments/assets/a1e99b90-3c1f-44ac-a9b8-3d0a34c52795">

Though I'm sure production data would be a bit different!

As far as I can tell, this is the only code that uses `CacheService.cached_data_by_class`, so there could be an opportunity to simplify the CacheService if the EnterpriseInjectionData no longer needed it

Reply:
Ah, I see. It seems pointless making an extra DB query to find out if we need to make the real query, when we could just make the real query. Perhaps with further optimisation, the real query might be quite efficient (I just added a comment to `Spree::Taxon.supplied_taxons`)